src/YourEasyRent/Services/TgButtonCallback.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Microsoft.AspNetCore.SignalR;
+using Microsoft.VisualBasic;
+using System.Numerics;
+using Telegram.Bot.Types;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace YourEasyRent.Services
+{
+    public class TgButtonCallback
+    {
+        private readonly Update _update;
+        public bool IsStart { get; set; }
+        public bool IsValidMessage {  get; set; }
+        public bool IsValueMenuMessage { get; set; } 
+        public bool IsBrandMenu {  get; set; }  
+        public bool IsCategoryMenu { get; set; }    
+        public bool IsValueProductButton {  get; set; }
+
+
+        public TgButtonCallback(Update update)
+        {
+            _update = update;
+        }
+
+        public void IsBotStart()
+        {
+
+            var messageText = _update.Message?.Text;
+            var nameOfButton = _update.CallbackQuery?.Data;
+            if (messageText != null && messageText.Contains("/start") || nameOfButton == "StartNewSearch") 
+            { 
+                IsStart = true;
+            }
+            else
+            {
+                IsStart = false;
+            }
+        }
+        public string GetUserId()
+        {
+            var userId = _update.CallbackQuery?.From.Id.ToString();
+            if (userId == null)
+            {
+                var startedForUserId = _update.Message?.From.Id.ToString();
+                return startedForUserId;
+            }
+            return userId;
+        }
+        public string GetChatId()
+        {
+            var chatId = _update.Message?.Chat.Id.ToString();
+            if(chatId == null)
+            {
+                chatId = _update.CallbackQuery?.From.Id.ToString();
+                return chatId;
+            }
+            return chatId;
+        }
+
+        public void IsValidMsg() 
+        {
+            var nameOfButton = _update.CallbackQuery?.Data;
```

Comment:
You can do 
`           
            var nameOfButton = _update.CallbackQuery?.Data **??** _update.Message?.Text **??** throw new Exception("The user did not send a message");;`

Reply:
did it